bin/install

```diff
@@ -38,23 +37,24 @@ _get_download_url() {
 install() {
 	local -r version=$1
 	local -r install_path=$2
-	local -r tar_install_path="$install_path/shellcheck-v$version.tgz"
-	local -r bin_install_path="$install_path/bin"
+
+	local -r bin_install_path="${install_path}/bin"
+	local -r bin_path="${bin_install_path}/shellcheck"
+
 	local -r platform="$(_get_platform)"
 	local arch; arch="$(_get_arch)"
 	# Currently there is no darwin/arm support: https://github.com/koalaman/shellcheck/releases
 	# If/when it is added, remove the folling line and make arch a read-only variable.
 	test "$platform" == "darwin" && arch="x86_64"
-	local -r download_url="$(_get_download_url "$version" "$platform" "$arch")"
-	local -r bin_path="$bin_install_path/shellcheck"
-
-	mkdir -p "$bin_install_path"
-	echo "Downloading shellcheck from $download_url"
-	curl -Ls "$download_url" -o "$tar_install_path"
-	tar -xf "$tar_install_path"
-	cp "shellcheck-v$version/shellcheck" "$bin_path"
-	rm -rf "shellcheck-v$version"
-	chmod +x "$bin_path"
+
+	local -r download_url="$(_get_download_url "${version}" "${platform}" "${arch}")"
+
+	mkdir -pv "${bin_install_path}"
+
+	echo "Downloading shellcheck from ${download_url} to ${bin_install_path}"
+	# curl to tar, without files on disk
+	( curl -Ls "${download_url}" | tar -xJv --strip-components=1 -C "${bin_install_path}" "shellcheck-v${version}/shellcheck" )
```

Comment:
why is necessary subshell here?

Reply:
It probably isn't necessary for functionality, and should work the same with or without it.

Reply:
Removed in latest commit.